include/zephyr/senss/senss_datatypes.h

```diff
@@ -0,0 +1,341 @@
+ * aligned the HID spec, using the format v*10^x to present the decimal value,
+ * where the v is integer number, either int8/uint8, int16/uint6, or int32/uint32, depends on
+ * required sensor data precision.
+ *
+ * The scale unit exponent x is int8 type int8 type with encoding meanings (page 68 of
+ * HID spec https://usb.org/sites/default/files/hutrr39b_0.pdf):
+ *
+ * Unit Exponent argument       Power of Ten (Scientific Notation)
+ * 0x00			       1 * 10E0
+ * 0x01			       1 * 10E1
+ * 0x02			       1 * 10E2
+ * 0x03			       1 * 10E3
+ * 0x04			       1 * 10E4
+ * 0x05			       1 * 10E5
+ * 0x06			       1 * 10E6
+ * 0x07			       1 * 10E7
+ * 0x08			       1 * 10E-8
+ * 0x09			       1 * 10E-7
+ * 0x0A			       1 * 10E-6
+ * 0x0B			       1 * 10E-5
+ * 0x0C			       1 * 10E-4
+ * 0x0D			       1 * 10E-3
+ * 0x0E			       1 * 10E-2
+ * 0x0F			       1 * 10E-1
+ *
+ * So, we can have below data present ranges:
+ *
+ * int8 type v:    [-128, 127] *10^[-8, 7]
+ * uint8 type v:   [0,	255] * 10^[-8, 7]
+ * int16 type v:   [-32768, 32767] * 10^[-8, 7]
+ * uint16 type v:  [0,	65535] * 10^[-8, 7]
+ * int32 type v:   [-2147483648,  2147483647] * 10^[-8, 7]
+ * uint32 type v:  [0,	4294967295] * 10^[-8, 7]
+ * int64 type v:   [-9223372036854775808,  9223372036854775807] * 10^[-8, 7]
+ * uint64 type v:  [0,	18446744073709551615] * 10^[-8, 7]
+ *
+ * To simple the data structure definition and save store memory, only keep v in code definitions,
+ * scale exponent x will defined in doc and spec,  but not explicitly present in code, for scenarios
+ * which need transfer to decimal value, such as in a algorithm process, need base on the sensor
+ * type and according the doc/spec to get the right scale exponent value x.
+ *
+ * A example in doc and spec can be like:
+ * 3D accelerometer:
+ *	Data Fields    Type	Unit	    Unit Exponent   Typical Range     Description
+ *	data[0]        int32	micro g     -6		    +/-4*10^6	      x axis acceleration
+ *	data[1]        int32	micro g     -6		    +/-4*10^6	      y axis acceleration
+ *	data[2]        int32	micro g     -6		    +/-4*10^6	      z axis acceleration
+ *
+ * Ambient Light:
+ *	Data Fields    Type	Unit	    Unit Exponent  Typical Range     Description
+ *	data[0]        uint32	milli lux   -3		   [0, 10000] *10^3  Ambient light lux level
+ *
+ * The complete doc/spec should describe all supported sensors like above example.
+ */
+
+/**
+ * @brief Sensor value data structure for quaternion.
+ * int16 version.
+ */
+struct senss_sensor_value_quaternion_int16 {
+	struct senss_sensor_value_header header;
+	struct {
+		uint32_t timestamp_delta;
+		int16_t  x;
+		int16_t  y;
+		int16_t  z;
+		int16_t  w;
+	} readings[1];
+};
+
+/**
+ * @brief Sensor value data structure for 3-axis sensors.
+ * int32 version
+ */
+struct senss_sensor_value_3d_int32 {
+	struct senss_sensor_value_header header;
+	struct {
+		uint32_t timestamp_delta;
```

Comment:
Can we move the delta into the header as `interval`? This makes it very difficult to be able to do large operations on the `readings[]` array.

Reply:
according to the agreement on https://github.com/zephyrproject-rtos/zephyr/pull/55389#discussion_r1198403600 will keep the timestamp_delta in each reading.